conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -454,26 +455,32 @@ class zetMetricQueryTest : public zetMetricGroupTest {
     metricQueryHandle = lzt::metric_query_create(metricQueryPoolHandle);
   }
   void TearDown() override {
-    lzt::destroy_metric_query(metricQueryHandle);
-    lzt::destroy_metric_query_pool(metricQueryPoolHandle);
+    if (metricQueryHandle != nullptr) {
+      lzt::destroy_metric_query(metricQueryHandle);
+    }
+    if (metricQueryPoolHandle != nullptr) {
+      lzt::destroy_metric_query_pool(metricQueryPoolHandle);
+    }
   }
 };
 
 TEST_F(
     zetMetricQueryTest,
     GivenValidMetricQueryPoolWhenValidMetricGroupIsPassedThenExpectQueryhandle) {
+  zetMetricQueryInitialize();
```

Comment:
would it make sense to make it be called in the setup() of the zetMetricQueryTest fixture ?

Reply:
meaning, the zetMetricQueryTest  setup() looks for query metric groups. The starting of the test checks the list is not empty.

Reply:
Resolved, Thank you.